packages/nimble-components/src/wafer-map/index.ts

```diff
@@ -153,34 +155,88 @@ export class WaferMap extends FoundationElement {
     /**
      * @internal
      */
-    public render(): void {
-        this.renderQueued = false;
-        this.initializeInternalModules();
-        this.renderer?.drawWafer();
+    public setHoverData(
+        hoverWidth: number,
+        hoverHeight: number,
+        hoverOpacity: HoverDieOpacity,
+        hoverTransform: string
+    ): void {
+        this.hoverWidth = hoverWidth;
+        this.hoverHeight = hoverHeight;
+        this.hoverOpacity = hoverOpacity;
+        this.hoverTransform = hoverTransform;
     }
 
-    private queueRender(): void {
-        if (!this.$fastController.isConnected) {
-            return;
-        }
-        if (!this.renderQueued) {
-            this.renderQueued = true;
-            DOM.queueUpdate(() => this.render());
+    /**
+     * @internal
+     * Update function called when an update is queued.
+     * It will check which updates are needed based on which properties have changed.
+     * Each update represents a different starting point of the same sequential update flow.
+     * The updates snowball one after the other, this function only choses the 'altitude'.
+     * The hover does not require an event update, but it's also the last update in the sequence.
+     */
+    public update(): void {
+        if (this.waferMapUpdateTracker.requiresEventsUpdate) {
+            this.eventCoordinator.detachEvents();
+            if (this.waferMapUpdateTracker.requiresContainerDimensionsUpdate) {
+                this.updateContainerDimensions();
+            } else if (this.waferMapUpdateTracker.requiresScalesUpdate) {
+                this.updateScales();
+            } else if (
+                this.waferMapUpdateTracker.requiresLabelsFontSizeUpdate
+            ) {
+                this.updateLabelsFontSize();
+            } else if (
+                this.waferMapUpdateTracker.requiresDiesRenderInfoUpdate
+            ) {
+                this.updateDiesRenderInfo();
+            } else if (
+                this.waferMapUpdateTracker.requiresRenderingModuleUpdate
+            ) {
+                this.updateRenderingModule();
+            }
```

Comment:
so every single one of these code paths ends up calling `renderer.drawWafer`. I don't think the indirection is making it clearer what work this method is doing. Did you have something in mind for needing all the indirection? Would something like the following be clearer?


```suggestion
            if (this.waferMapUpdateTracker.requiresContainerDimensionsUpdate) {
                this.dataManager.updateContainerDimensions();
            } else if (this.waferMapUpdateTracker.requiresScalesUpdate) {
                this.dataManager.updateScales();
            } else if (
                this.waferMapUpdateTracker.requiresLabelsFontSizeUpdate
            ) {
                this.dataManager.updateLabelsFontSize();
            } else if (
                this.waferMapUpdateTracker.requiresDiesRenderInfoUpdate
            ) {
                this.dataManager.updateDiesRenderInfo();
            }
            this.renderer.drawWafer();
```

Reply:
It is something that remained from before the previous change, when the drawWafer was not always the last call. Thaks for noticing!

Reply:
Unresolving because the second part of my concern based on the proposed snippet was not addresses.
The question is are all these one line wrapper indirections still needed, notice in my snippet above that they were inlined instead:

https://github.com/ni/nimble/blob/e2a99c375d20a544bd85567f9dae473dafcc8b98/packages/nimble-components/src/wafer-map/index.ts#L201-L219